syntract_viewer/core.py

```
@@ -1191,11 +1174,39 @@ def _generate_and_apply_high_density_mask_coronal(nifti_file, trk_file, output_f
         if debug:
             print(f"  Smoothing complete")
     
+    # CRITICAL: Apply white mask as POST-PROCESSING to remove leaked areas
+    # This ensures mask dilation/smoothing doesn't spread into non-WM regions
+    if white_mask_slice is not None:
+        # Multiply final mask by white mask to strictly constrain to WM
+        mask_before = np.sum(high_density_mask > 0)
+        high_density_mask = high_density_mask * white_mask_slice
+        mask_after = np.sum(high_density_mask > 0)
+        if debug:
+            print(f"  Applied white mask post-processing filter")
```

Ruff (0.15.21)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)